### PR TITLE
Custom location sharing.

### PR DESCRIPTION
**Pull request by ara4n**
Builds on top of #7135; adds in basic support for sharing custom locations.

Screenshot 2021-11-23 at 00 45 35

---
Here's what your changelog entry will look like:

## ✨ Features
 * Custom location sharing. ([\#7185](https://github.com/matrix-org/matrix-react-sdk/pull/7185)).

Preview: https://61ade198607f187b7ee9fda2--matrix-react-sdk.netlify.app
⚠️ Do you trust the author of this PR? Maybe this build will steal your keys or give you malware. Exercise caution. Use test accounts.

### CODE REVIEW

**Comment by t3chguy** [comment]
(blocked on stacked PR)

File: src/components/views/location/LocationPicker.tsx [hunk 1]
```diff
@@ -114,9 +115,47 @@ class LocationPicker extends React.Component<IProps, IState> {
             this.geolocate.trigger();
         });
 
+        this.map.on('click', (e)=>{
```
Comment (t3chguy) [inline src/components/views/location/LocationPicker.tsx:118]:
```suggestion
        this.map.on('click', (e) => {
```

File: src/components/views/location/LocationPicker.tsx [hunk 2]
```diff
@@ -114,9 +115,47 @@ class LocationPicker extends React.Component<IProps, IState> {
...
+            this.addMarker(e.lngLat);
+            this.storeManualPosition(e.lngLat);
+            this.setState({ type: LocationShareType.Custom });
+        });
+
         this.geolocate.on('geolocate', this.onGeolocate);
     }
 
+    private addMarker(lngLat) {
```
Comment (t3chguy) [inline src/components/views/location/LocationPicker.tsx:127]: type?

File: src/components/views/location/LocationPicker.tsx [hunk 3]
```diff
@@ -114,9 +115,47 @@ class LocationPicker extends React.Component<IProps, IState> {
...
+            .on('dragend', ()=>{ this.storeManualPosition(this.marker.getLngLat()); });
+    }
+
+    private removeMarker() {
+        if (!this.marker) return;
+        this.marker.remove();
+        this.marker = undefined;
+    }
+
+    private storeManualPosition(lngLat) {
```
Comment (t3chguy) [inline src/components/views/location/LocationPicker.tsx:143]: type?

File: src/components/views/messages/MLocationBody.tsx [hunk 4]
```diff
@@ -67,15 +67,26 @@ export default class MLocationBody extends React.Component<IBodyProps> {
...
-            center: [this.coords.longitude, this.coords.latitude],
+            center: coordinates,
             zoom: 13,
         });
 
-        new maplibregl.Marker()
-            .setLngLat([this.coords.longitude, this.coords.latitude])
+        // const marker = new maplibregl.Marker()
+        //     .setLngLat(coordinates)
+        //     .addTo(this.map);
```
Comment (t3chguy) [inline src/components/views/messages/MLocationBody.tsx:81]: any reason for this to live?

**Comment by ara4n** [comment]
marking as unblocked given #7135 is (hopefully) sorted now.

**Review by t3chguy — APPROVED** [review]

**Pull request merged**